src/js/core/factories/GridScrolling.js

```diff
@@ -0,0 +1,300 @@
+(function() {
+  'use strict';
+
+  angular.module('ui.grid')
+    .factory('gridScrolling', ['$window', 'gridUtil', 'uiGridConstants',
+      function($window, gridUtil, uiGridConstants) {
+        var isAnimating;
+
+        /**
+         *  @ngdoc object
+         *  @name initiated
+         *  @propertyOf  ui.grid.class:gridScrolling
+         *  @description Keeps track of which type of scrolling event has been initiated
+         *  and sets it to NONE, when no event is being triggered.
+         */
+        gridScrolling.initiated = uiGridConstants.scrollType.NONE;
+
+        /**
+         * @ngdoc function
+         * @name ui.grid.class:gridScrolling
+         * @description gridScrolling is a wrapper service that takes over the default scrolling logic in order to
+         * ensure that grid scrolling works consistently in both the browser and devices, as well as slow machines.
+         * @param {object} element Element being scrolled
+         * @param {function} scrollHandler Function that needs to be called when scrolling happens.
+         */
+        function gridScrolling(element, scrollHandler) {
+          var wrapper = element, pointX, pointY, startTime, startX, startY, maxScroll,
```

Comment:
That is a good question. I didn't catch that when I was refactoring it. Will change it.